nncf/experimental/openvino_native/quantization/algorithms/min_max/openvino_backend.py

```diff
@@ -172,32 +175,59 @@ def _get_num_samples(num_samples, target_point: OVTargetPoint):
 
-    @staticmethod
-    def mean_minmax_statistic_collector(nncf_graph: NNCFGraph,
-                                        target_point: OVTargetPoint,
-                                        quantizer_config: QuantizerConfig,
-                                        use_per_sample_stats: bool,
-                                        inplace: bool,
-                                        num_samples: int = None,
-                                        ) -> TensorCollector:
-        reduction_shape, use_abs_max = \
-            OVMinMaxAlgoBackend._get_reduction_shape_and_use_abs_max(nncf_graph, target_point,
-                                                                     quantizer_config)
-        _num_samples = OVMinMaxAlgoBackend._get_num_samples(num_samples, target_point)
-        return get_mean_min_max_stat_collector(_num_samples, reduction_shape,
-                                               use_abs_max, use_per_sample_stats, inplace)
+        try:
+            base_kwargs = {
+                'reduction_shape': reduction_shape,
+                'inplace': inplace
+            }
+            reducers = []
+            for i, params in enumerate([range_estimator_params.min, range_estimator_params.max]):
+                kwargs = base_kwargs.copy()
+                if params.statistics_type in [StatisticsType.QUANTILE,
+                                              StatisticsType.ABS_QUANTILE]:
+                    kwargs.update({'quantile': [i + (1 - 2 * i) * params.quantile_outlier_prob]})
+                # TODO(dlyakhov): merge two quantile aggregators in one
+                statistic_type = params.statistics_type
+                if use_abs_max and statistic_type == StatisticsType.MAX:
+                    statistic_type = StatisticsType.ABS_MAX
+
+                reducers.append(OV_REDUCERS_MAP[statistic_type](**kwargs))
+        except Exception as e:
+            raise Exception(
+                f'Could not build statistic functions:'
+                f' {range_estimator_params.min.statistics_type},'
+                f' {range_estimator_params.max.statistics_type}') from e
+
+        try:
+            kwargs = {
+                'num_samples': _num_samples,
+                'tensor_processor': OVNNCFCollectorTensorProcessor
+            }
+            aggregators = []
+            for params in [range_estimator_params.min, range_estimator_params.max]:
+                aggregator_cls = AGGREGATORS_MAP[params.aggregator_type]
+                aggregators.append(aggregator_cls(**kwargs))
+        except Exception as e:
+            raise Exception(
+                f'Could not build statistic aggregators:'
+                f' {range_estimator_params.min.aggregator_type},'
+                f' {range_estimator_params.max.aggregator_type}') from e
+
+        collector = TensorCollector(OVMinMaxTensorStatistic)
+        collector.register_statistic_branch(OVMinMaxTensorStatistic.MIN_STAT, reducers[0], aggregators[0])
```

Comment:
Why do we collect `reducers` & `aggregators` in a loop and then use them only by indexes?

Reply:
Just to avoid code duplication for min and max redcuers/aggregators

Reply:
I can register branches in a loop as well, that should be convenient

Reply:
Please check